atom/browser/atom_browser_main_parts.cc

```diff
@@ -183,6 +184,8 @@ void AtomBrowserMainParts::PreMainMessageLoopRun() {
   std::unique_ptr<base::DictionaryValue> empty_info(new base::DictionaryValue);
   Browser::Get()->DidFinishLaunching(*empty_info);
 #endif
+
+  atom::api::App::Get()->PreMainMessageLoopRun();
```

Comment:
For consistent behavior on different platforms, this should be put before the `WillFinishLaunching ` call, so the actual listening starts before the `ready` event.

Reply:
For clean architecture the classes under `api` namespace usually should not be accessed by the outside namespace, otherwise the browser initialization code would have to know when the module system starts.

A cleaner design to achieve this is to add `PreMainMessageLoopRun` to `BrowserObserver` and then call `Browser::Get()->PreMainMessageLoopRun()`.